sourcegraph/conn_pool.go

```diff
@@ -41,7 +41,7 @@ func pooledGRPCDial(target string, opts ...grpc.DialOption) (*grpc.ClientConn, e
 	if conns == nil {
 		conns = map[string]*grpc.ClientConn{}
 	}
-	if conn := conns[target]; conn != nil {
+	if conn := conns[target]; conn != nil && conn.State() != grpc.Shutdown {
```

Comment:
This additional check will ignore `conn` objects that are in a `Shutdown` state and call `grpc.Dial` again for that target. This let's the client reboot a connection in case the upstream server ditched the connection.